Observer/SalesQuoteProductAddAfter.php

```diff
@@ -0,0 +1,157 @@
+            $this->klAddedToCartItemData( $quote, $item );
+        }
+    }
+
+    /**
+     * Build Added to Cart payload for current Added item
+     * @param $quote
+     * @param $addedItem
+     */
+    public function klAddedToCartItemData( $quote, $addedItem )
+    {
+        $addedProduct = $addedItem->getProduct();
+        $addedItemData = array(
+            'AddedItemCategories' => (array) $this->getCategoryName( $addedProduct->getCategoryIds() ),
+            'AddedItemDescription' => (string) strip_tags( $addedProduct->getDescription() ),
+            'AddedItemImageUrlKey' => (string) $addedProduct->getData('small_image'),
+            'AddedItemPrice' => (float) $addedProduct->get_price(),
+            'AddedItemQuantity' => (int) $addedItem->getQty(),
+            'AddedItemProductID' => (int) $addedProduct->getId(),
+            'AddedItemProductName' => (string) $addedItem->getName(),
+            'AddedItemSku' => (string) $addedProduct->getSku(),
+            'AddedItemUrl' => (string) $addedProduct->getProductUrl()
+        );
+
+        $klAddedToCartPayload = array_merge( $this->klBuildCartData( $quote, $addedItem ), $addedItemData );
+
+        $this->getCheckoutSession()->setKlAddedToCartKey( $klAddedToCartPayload );
+    }
+
+    /**
+     * Helper function to add items from cart to Added to Cart payload
+     * @param $quote
+     * @param $addedItem
+     * @return array
+     */
+    public function klBuildCartData( $quote, $addedItem )
+    {
+        $cartItems = $quote->getItems() ? $quote->getitems() : array();
+        $cartQty = 0;
+        $items = array();
+        $cartItemNames = array();
+        $cartItemCategories = array();
+
+        foreach( $cartItems as $item ) {
+            $product = $item->getProduct();
+            $cartItemId = $product->getId();
+            $itemCategories = $this->getCategoryName( $product->getCategoryIds() );
+            $itemName = $item->getName();
+            $currentProduct = array(
+                'Categories' => (array) $itemCategories,
+                'ImageUrlKey' => (string) $product->getData('small_image'),
+                'ProductId' => (int) $cartItemId,
+                'Price' => (float) $product->getPrice(),
+                'Title' => (string) $itemName,
+                'Description' => (string) strip_tags( $product->getDescription() ),
+                'Url' => (string) $product->getProductUrl(),
+                'Quantity' => (int) $item->getQty()
+            );
+            $cartQty += $item->getQty();
+            array_push( $items, $currentProduct );
+            array_push( $cartItemNames, $itemName );
+            $cartItemCategories = $this->uniqueArrayOfStrings( $cartItemCategories, $itemCategories );
+        }
+
+        return array(
+            '$value' => (float) $quote->getBaseGrandTotal() + $addedItem->getPrice(),
+            'ItemNames' => (array) $cartItemNames,
+            'Items' => (array) $items,
```

Comment:
I think the event structure should probably mimic the Checkout Started event so that it's quick and easy to repurpose a flow template for an Abandoned Cart flow triggered ont he Added to Cart event. The array of cart line items should be under `event.Extra.LineItems`